bin/ncp-update

```diff
@@ -20,7 +20,159 @@
   [[ -f /.ncp-image ]] || cd "$TMPDIR"         # update locally during build
 
   echo -e "Performing updates"
-  ./update.sh || exit 1
+
+  # Latest checkpoint is the version right before the latest cleanup of update.sh
+  # Get the array of updates dir
+  # The files in updates dir are sorted by tag number
+  i=0
+  while read line ; do
+    updates_list[ $i ]="$line"
+    (( i++ ))
+  done < <( ls -1 updates | sort -V)
+
+  starting_checkpoint=0
+  len=${#updates_list[@]}
+  end_of_list=$(expr $len - 1)
+
+  # The latest checkpoint is the newer version in updates dir
+  latest_checkpoint=${updates_list[$end_of_list]}
+
+  # Compare current version with latest checkpoint to see if we need backwards updates
```

Comment:
let's make a function in `etc/library.sh` to that returns true if version A is more recent than version B